src/riak_repl2_fscoordinator.erl

```diff
@@ -566,9 +566,10 @@ handle_abnormal_exit(ExitType, Pid, _Cause, {value, PartitionWithSource, Running
                     Dropped = [Partition#partition_info.index | State4#state.dropped],
                     Purgatory = queue:filter(fun({P, _}) -> P =/= Partition end,
                                              State4#state.purgatory),
-                    {noreply, State4#state{error_exits = ErrorExits1,
-                                           purgatory = Purgatory,
-                                           dropped = Dropped}};
+                    maybe_complete_fullsync(Running,
```

Comment:
This is more than just a refactoring...

Reply:
You are right, I was a bit lazy combining the two.  Shall I revert out the commits associated with https://github.com/basho/riak_repl/issues/799 and put in a separate PR?

Reply:
Thanks for the reference. This is fine.